deliver/lib/deliver/upload_trailers.rb

```diff
@@ -0,0 +1,94 @@
+require 'pry'
+module Deliver
+  # upload video trailer to ITC
+  class UploadTrailers
+    def upload(options, videos)
+      return if options[:skip_trailers]
+
+      app = options[:app]
+
+      v = app.edit_version
+      UI.user_error!("Could not find a version to edit for app '#{app.name}'") unless v
+
+      # We have to make sure that it's only one per device in lang
+      indexed = {}
+
+      UI.message("Uploading #{videos.length} video trailers")
+
+      # Opposite to screenshots, video trailers do not divide per language.
+      # Trailer uploaded for one device and one language will propagate on all active languages.
+      # Though ITC will return trailers divided per language (v.trailers method). Don't be fooled. It doesn't mean anything.
+      # You cant set two different trailers for the same device on two languages
+      # ITC you are so consistent...
+      videos.each do |trailer|
```

Comment:
Does this logic work if you want to remove all trailers ? I.e. you have no trailer locally, it won't delete any ?